src/main/java/de/tum/cit/aet/artemis/communication/web/ForwardedMessageResource.java

```diff
@@ -0,0 +1,102 @@
+
+import de.tum.cit.aet.artemis.communication.domain.ForwardedMessage;
+import de.tum.cit.aet.artemis.communication.dto.ForwardedMessageDTO;
+import de.tum.cit.aet.artemis.communication.repository.ForwardedMessageRepository;
+import de.tum.cit.aet.artemis.core.exception.BadRequestAlertException;
+
+/**
+ * REST controller for managing ForwardedMessages.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping(value = "api")
+public class ForwardedMessageResource {
+
+    private static final Logger log = LoggerFactory.getLogger(ForwardedMessageResource.class);
+
+    private static final String ENTITY_NAME = "forwardedMessage";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final ForwardedMessageRepository forwardedMessageRepository;
+
+    public ForwardedMessageResource(ForwardedMessageRepository forwardedMessageRepository) {
+        this.forwardedMessageRepository = forwardedMessageRepository;
+    }
+
+    /**
+     * POST /forwarded-messages : Create a new forwarded message.
+     *
+     * @param forwardedMessage the forwarded message to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new forwarded message,
+     *         or with status 400 (Bad Request) if the forwarded message has already an ID
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/forwarded-messages")
+    public ResponseEntity<ForwardedMessageDTO> createForwardedMessage(@RequestBody ForwardedMessage forwardedMessage) throws URISyntaxException {
+        log.debug("REST request to save ForwardedMessage : {}");
+        if (forwardedMessage.getId() != null) {
+            throw new BadRequestAlertException("A new forwarded message cannot already have an ID", ENTITY_NAME, "idExists");
+        }
+        ForwardedMessage savedMessage = forwardedMessageRepository.save(forwardedMessage);
+        ForwardedMessageDTO dto = new ForwardedMessageDTO(savedMessage);
+        return ResponseEntity.created(new URI("/api/forwarded-messages/" + savedMessage.getId())).body(dto);
+    }
+
+    /**
+     * GET /forwarded-messages/posts : Retrieve forwarded messages grouped by their destination post IDs.
+     *
+     * @param dest_post_ids a set of destination post IDs for which the forwarded messages should be retrieved
+     * @return the ResponseEntity with status 200 (OK) and with body containing a map where the key is the destination post ID
+     *         and the value is a set of forwarded message DTOs grouped by the destination post IDs
+     */
+    @GetMapping("/forwarded-messages/posts")
+    public ResponseEntity<Map<Long, Set<ForwardedMessageDTO>>> getForwardedMessagesByDestPostIds(@RequestParam Set<Long> dest_post_ids) {
+        Set<ForwardedMessage> forwardedMessages = forwardedMessageRepository.findAllByDestinationPostIds(dest_post_ids);
+
+        Map<Long, Set<ForwardedMessageDTO>> groupedDtos = forwardedMessages.stream()
```

Comment:
do you really need to return a datatype like this? `Map<Long, Set<ForwardedMessageDTO>>`. This could also be a list of a DTO with `(id, Set<ForwardedMessageDTO)`